src/input/json_to_storage.cc

```diff
@@ -336,21 +336,37 @@ StorageBase * JSONToStorage::make_storage(JSONPath &p, const Type::Record *recor
 {
     if (p.head()->type() == json_spirit::obj_type) {
     	const json_spirit::mObject & j_map = p.head()->get_obj();
-    	std::set<string> keys_to_processed;
+    	std::set<string> keys_to_process;
         json_spirit::mObject::const_iterator map_it;
         std::set<string>::iterator set_it;
 
         for( map_it = j_map.begin(); map_it != j_map.end(); ++map_it) {
-           keys_to_processed.insert(map_it->first);
+           keys_to_process.insert(map_it->first);
+        }
+
+        if (keys_to_process.find("TYPE") != keys_to_process.end()) {
+            string value;
+            string ref_address;
+            p.down( record->key_iterator("TYPE")->key_ );
```

Comment:
Rather use same approach as in AbstractRecord. Use try block and catch exception for unknown selection key.
Need not to deal with ref_path etc.